python/grass/tools/session_tools.py

```diff
@@ -390,12 +446,42 @@ def run_cmd(
         :param tool_kwargs: named tool arguments used for error reporting (experimental)
         :param **popen_options: additional options for :py:func:`subprocess.Popen`
         """
-        return self.call_cmd(
-            command,
-            tool_kwargs=tool_kwargs,
-            input=input,
-            **popen_options,
-        )
+        # Compute the environment for subprocesses and store it for later use.
+        if "env" not in popen_options:
+            popen_options["env"] = self._modified_env_if_needed()
+
+        if parameter_converter is None:
```

Comment:
This section could use some comments.

Reply:
I made the function protected and created a wrapper for the API. This helps to give context to the parameters.

I added comments to the function code.